substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/JavaUtilZipSubstitutions.java

```diff
@@ -154,26 +154,34 @@ private static long init(int level, int strategy, boolean nowrap) {
         }
     }
 
+    private static void doSetDictionary(long addr, CCharPointer buf, int len) {
```

Comment:
I usually like to put in a link to where the original source code is.  Like the minimalist comment at line 132.  The idea is to distinguish translations from new code (e.g., utility methods) that you write.

I would expand that to `src/java.base/share/native/libzip/Deflater.c` and hope that they don't rearrange the code base (again :-).

Reply:
All fixed.  I noted what JDK version the copied code came from as well.